tests/unit/controllers/admin/seeder-test.js

```diff
@@ -1,119 +1,113 @@
+    controller._destroyAll = stub().resolves();
+    controller.model = { libraries: 'libraryRecords' };
+
+    controller.send('deleteLibraries');
+
+    await assert.equal(controller.deleteLibrariesInProgress, true);
+    await assert.ok(controller._destroyAll.calledOnceWith('libraryRecords'));
+    await assert.equal(controller.libDelDone, true);
+    await assert.equal(controller.deleteLibrariesInProgress, false);
   });
 
-  test('generateBooksAndAuthors action', function(assert) {
-    const done = assert.async();
-    const { controller } = this;
-    controller.send('generateBooksAndAuthors', 10);
+  test('generateBooksAndAuthors action', async function(assert) {
     assert.expect(5);
-    assert.equal(controller.get('generateBooksInProgress'), true);
-    assert.equal(controller._saveRandomAuthor.callCount, 10);
-
-    next(() => {
-      assert.ok(controller._generateSomeBooks.calledWith('newAuthor'));
-      assert.equal(controller.get('authDone'), true);
-      assert.equal(controller.get('generateBooksInProgress'), false);
-      done();
+
+    const controller = this.owner.lookup('controller:admin/seeder');
+    controller._saveRandomAuthor = stub().resolves('newAuthor');
+    controller._generateSomeBooks = stub().resolves();
+
+    controller.send('generateBooksAndAuthors', 10);
+
+    await assert.equal(controller.generateBooksInProgress, true);
+    await assert.equal(controller._saveRandomAuthor.callCount, 10);
+    await assert.ok(controller._generateSomeBooks.calledWith('newAuthor'));
+
+    next(async () => {
+      await assert.equal(controller.authDone, true);
+      await assert.equal(controller.generateBooksInProgress, false);
     });
   });
 
-  test('deleteBooksAndAuthors action', function(assert) {
-    const done = assert.async();
-    const { controller } = this;
+  test('deleteBooksAndAuthors action', async function(assert) {
+    assert.expect(5);
+
+    const controller = this.owner.lookup('controller:admin/seeder');
+    controller._destroyAll = stub().resolves();
+    controller.authors = 'authorRecords';
+    controller.books = 'bookRecords';
+
     controller.send('deleteBooksAndAuthors');
-    assert.expect(4);
-    assert.equal(controller.get('deleteBooksInProgress'), true);
 
-    next(() => {
-      assert.deepEqual(controller._destroyAll.args, [['authors'], ['books']]);
-      assert.equal(controller.get('authDelDone'), true);
-      assert.equal(controller.get('deleteBooksInProgress'), false);
-      done();
-    });
-  });
+    await assert.equal(controller.deleteBooksInProgress, true);
 
-  module('private methods', function(hooks) {
-    hooks.beforeEach(function() {
-      const save = spy();
-      const randomize = stub().returns({ save });
-      const createRecord = stub().returns({ randomize });
-      const controller = {
-        store: { createRecord },
-      };
-      this.controller = this.owner.factoryFor('controller:admin/seeder').create(controller);
+    next(async () => {
```

Comment:
weird, but here we have to use next otherwise promises not managed properly...

Reply:
Yeah, we should not need two async patterns. Definitely something to revisit later. I created an issue (#74 .)